xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/ComputedFieldClass.java

```diff
@@ -90,6 +90,47 @@ public String getScript()
         return sValue;
     }
 
+    /**
+     * Computes and returns the raw value of this property for a given object.
+     * @param name property name
+     * @param prefix prefix to be added
+     * @param object object for which the property value has to get computed
+     * @param context current context
+     * @return the computed property value
+     */
+    public String getComputedValue(String name, String prefix, BaseCollection object, XWikiContext context) {
+        String script = getScript();
+
+        try {
+            ScriptContext scontext = Utils.getComponent(ScriptContextManager.class).getCurrentScriptContext();
+            scontext.setAttribute("name", name, ScriptContext.ENGINE_SCOPE);
+            scontext.setAttribute("prefix", prefix, ScriptContext.ENGINE_SCOPE);
+            scontext.setAttribute("object", new com.xpn.xwiki.api.Object((BaseObject) object, context),
+                    ScriptContext.ENGINE_SCOPE);
+
+            XWikiDocument classDocument = object.getXClass(context).getOwnerDocument();
+
+            return renderContentInContext(script, classDocument.getSyntax().toIdString(),
+                    classDocument.getAuthorReference(), classDocument.getDocumentReference(), context);
+
+        } catch (Exception e) {
+            return e.getMessage();
```

Comment:
You should keep the previous TODO since just returning the exception message is not so great.

Reply:
Indeed, I had removed it by mistake, I restored it.

Reply:
I would throw an XWikiException here, and keep this with the todo in a try/catch of `displayView` then you should be able to fallback in case of problem in `ModelFactory`. Right now, you risk to put a wrong information in the field in case of error.

Reply:
Thanks surli, I understand the risk of displaying wrong information is an issue, I tried to get my head around your suggestion but I'm not sure how to fallback. Also, the method PropertyClass:renderContentInContext throws Exception, not XWikiException so I'm wondering what to throw. I brought some changes, please let me know what you think.